File: tools/python-incident-analyzer/src/incident_analyzer/domain/incident_status.py
"""Incident lifecycle status."""

from __future__ import annotations

from enum import StrEnum


class IncidentStatus(StrEnum):
    """Incident lifecycle status mapped from the Java backend."""

    OPEN = "OPEN"
    INVESTIGATING = "INVESTIGATING"
    RESOLVED = "RESOLVED"

    @classmethod
    def from_json(cls, value: str) -> IncidentStatus:
        """Parse a backend status value into an enum."""
        try:
            return cls(value)
        except ValueError as exc:
            raise ValueError(f"Unknown incident status: {value}") from exc

File: frontend/flutter-rules-practice/lib/features/incidents/domain/incident_status.dart
/// Incident lifecycle status.
///
/// Values are mapped from the backend enum format:
/// `OPEN`, `INVESTIGATING`, `RESOLVED`.
enum IncidentStatus {
  open,
  investigating,
  resolved;

  /// Parses the backend enum value into a Dart enum.
  static IncidentStatus fromJson(String value) {
    return switch (value) {
      'OPEN' => IncidentStatus.open,
      'INVESTIGATING' => IncidentStatus.investigating,
      'RESOLVED' => IncidentStatus.resolved,
      _ => throw ArgumentError('Unknown incident status: $value'),
    };
  }

  String toJson() {
    return switch (this) {
      IncidentStatus.open => 'OPEN',
      IncidentStatus.investigating => 'INVESTIGATING',
      IncidentStatus.resolved => 'RESOLVED',
    };
  }

  String get label {
    return switch (this) {
      IncidentStatus.open => 'Open',
      IncidentStatus.investigating => 'Investigating',
      IncidentStatus.resolved => 'Resolved',
    };
  }
}